### PR TITLE
feat(ButtonGroup): Added highlightSelected and justified prop

### PR DESCRIPTION
**Pull request by shashankcs090**
screen shot 2018-06-12 at 2 28 14 pm
@pauljeter @bfbiggs

### CODE REVIEW

File: scss/components/button-group.scss [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+@import '../tools/mixins/border-radius';
+@import '../tools/mixins/vendor-prefixes';
+
+@include exports('cui-button-group') {
+  .#{$button-group__class} {
+    display: flex;
+    width: 100%;
+
+    > .#{$button__class} {
+      background-color: transparent;
```
Comment (bfbiggs) [inline scss/components/button-group.scss]: Could button mixins be used here?

File: scss/components/button-group.scss [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    &.#{$button-group__class} {
+      &--justified {
+        > .#{$button__class}:not(.#{$button__class}--icon) {
+          flex-grow: 1;
+          width: 0;
+        }
+      }
+
+      &--space {
+        > .#{$button__class} {
```
Comment (bfbiggs) [inline scss/components/button-group.scss]: Could button mixins be used here?

Comment (shashankcs090) [reply scss/components/button-group.scss]: I'll see if I can use those.

**Comment by shashankcs090** [comment]
focus states also taken care off.

**Comment by bfbiggs** [comment]
```
/*
* @category containers
* @component button-group
* @design collaboration-design
*/

@import '../settings/buttons';
@import '../settings/button-group';
@import '../tools/mixins/border-radius';
@import '../tools/mixins/buttons';
@import '../tools/mixins/vendor-prefixes';

@include exports('cui-button-group') {
  .#{$button-group__class} {
    display: flex;
    width: 100%;

    > .#{$button__class} {

      @include button-color(
        $color: $button-group__color,
        $color-hover: $button-group__color,
        $color-active: $button-group-active__color,
        $color-disabled: $button-group-disabled__color,
        $bg: $button-group__background,
        $bg-active: $button-group-active__background,
        $bg-disabled: $button-group__background,
        $bg-hover:$button-group-hover__background
      );

      padding: $button-group__padding;
      font-size: $button-group__font-size;
      border-radius: 0;
      border: $button-group__border;
      line-height: $button-group__line-height;
      margin-left: 0;
      min-width: 0;

      &:first-child {
        @include border-left-radius($button-group__border-radius);
      }

      &:not(:first-child) {
        margin-left: -$button-group__border-size;
        border-left-color: $black-04;
      }

      &:not(:last-child) {
        border-right-color: $black-04;
      }


      &:last-child {
        @include border-right-radius($button-group__border-radius);
      }

      &.#{$button__class}--icon-group {
        padding: $button-group-icon__padding;
        width: auto;
        .cui-icon {
          fill: currentColor;
        }
      }
    }

    &.#{$button-group__class} {
      &--justified {
        > .#{$button__class}:not(.#{$button__class}--icon-group) {
          flex-grow: 1;
          width: 0;
        }
      }

      &--space {
        > .#{$button__class} {

          @include button-color(
            $color: $button-group__color--space,
            $color-hover: $button-group__color--space,
            $color-active: $button-group-active__color--space,
            $color-disabled: $button-group-disabled__color--space,
            $bg: $button-group__background,
            $bg-active: $button-group-active__background--space,
            $bg-disabled: $button-group__background,
            $bg-hover:$button-group-hover__background--space
          );
          border: $button-group__border--space;
          font-size: $button-group__font-size--space;
          line-height: $button-group__line-height--space;
          padding: $button-group__padding--space;

          &:not(:first-child) {
            border-left-color: $white-12;
          }
    
          &:not(:last-child) {
            border-right-color: $white-12;
          }

          &.active {
            font-weight: bold;
          }
        }
      }
    }
  }
}
```

**Pull request closed without merging**